### PR TITLE
Issue #61 Introducing a new Port Library API for obtaining open file …

### PR DESCRIPTION
**Pull request by sonchakr**
A new API is being added to the port library that returns the number of files opened.
Issue: #61
Signed-off-by: Sondhi Chakraborty <sondhi.chakraborty@in.ibm.com>

### CODE REVIEW

File: fvtest/porttest/si.cpp [hunk 1]
```diff
@@ -1845,4 +1947,75 @@ TEST(PortSysinfoTest, sysinfo_test_get_groups)
...
+#if defined(LINUX) || defined(AIXPPC)
+/**
+ * Test omrsysinfo_test_get_open_file_count.
+ * Available only on Linux and AIX.
+ */
+TEST(PortSysinfoTest, sysinfo_test_get_open_file_count)
+{
+	OMRPORT_ACCESS_FROM_OMRPORT(portTestEnv->getPortLibrary());
+	const char* testName = "omrsysinfo_test_get_open_file_count";
+    int32_t ret = 0;
```
Comment (alin00) [inline fvtest/porttest/si.cpp]: wrong indent

File: fvtest/porttest/si.cpp [hunk 2]
```diff
@@ -1845,4 +1947,75 @@ TEST(PortSysinfoTest, sysinfo_test_get_groups)
...
+	uint64_t openCount = 0;
+	uint64_t curLimit = 0;
+	uint32_t rc = 0;
+
+	reportTestEntry(OMRPORTLIB, testName);
+	/* Get the number of files opened till this point. */
+    ret = omrsysinfo_get_open_file_count(&openCount);
+    if (ret < 0) {
```
Comment (alin00) [inline fvtest/porttest/si.cpp]: wrong indent

File: fvtest/porttest/si.cpp [hunk 3]
```diff
@@ -1845,4 +1947,75 @@ TEST(PortSysinfoTest, sysinfo_test_get_groups)
...
+		outputErrorMessage(PORTTEST_ERROR_ARGS, "omrsysinfo_get_open_file_count() failed.\n");
+		reportTestExit(OMRPORTLIB, testName);
+		return;
+    }
```
Comment (alin00) [inline fvtest/porttest/si.cpp]: wrong indent

File: fvtest/porttest/si.cpp [hunk 4]
```diff
@@ -1845,4 +1947,75 @@ TEST(PortSysinfoTest, sysinfo_test_get_groups)
...
+
+#if defined(LINUX) || defined(AIXPPC)
+/**
+ * Test omrsysinfo_test_get_open_file_count.
+ * Available only on Linux and AIX.
+ */
+TEST(PortSysinfoTest, sysinfo_test_get_open_file_count)
+{
+	OMRPORT_ACCESS_FROM_OMRPORT(portTestEnv->getPortLibrary());
+	const char* testName = "omrsysinfo_test_get_open_file_count";
```
Comment (alin00) [inline fvtest/porttest/si.cpp]: should be ```const char *testName = ...```

File: port/unix/omrsysinfo.c [hunk 5]
```diff
@@ -2849,3 +2888,71 @@ setPortableError(OMRPortLibrary *portLibrary, const char *funcName, int32_t port
...
 }
+
+int32_t
+omrsysinfo_get_open_file_count(struct OMRPortLibrary *portLibrary, uint64_t *count)
+{
+	int32_t ret = 0;
+#if defined(LINUX) || defined(AIXPPC)
+	uint64_t fdCount = 0;
+	char buffer[PATH_MAX] = {0};
+	const char* procDirectory = "/proc/%d/fd/";
```
Comment (alin00) [inline port/unix/omrsysinfo.c]: should be ```const char *procDirectory = ...```

**Comment by alin00** [comment]
Please squash the commits.

**Comment by sonchakr** [comment]
Have addressed all comments from above - including and up to the last one - squashed all commits to 1.  Please review/initiate a pull.
Thank You.

**Comment by alin00** [comment]
Thank you. Merging.

**Pull request merged**